grails-geb/src/testFixtures/groovy/grails/plugin/geb/WebDriverContainerHolder.groovy

```diff
@@ -59,99 +64,130 @@ import grails.plugin.geb.serviceloader.ServiceRegistry
-        currentConfiguration = null
+        containerConf = null
     }
 
-    boolean matchesCurrentContainerConfiguration(WebDriverContainerConfiguration specConfiguration) {
-        specConfiguration == currentConfiguration && grailsGebSettings.recordingMode == BrowserWebDriverContainer.VncRecordingMode.SKIP
+    boolean matchesCurrentContainerConfiguration(WebDriverContainerConfiguration specConf) {
+        specConf == containerConf &&
+        settings.recordingMode == BrowserWebDriverContainer.VncRecordingMode.SKIP
     }
 
-    private static int getPort(IMethodInvocation invocation) {
+    private static int findServerPort(IMethodInvocation methodInvocation) {
         try {
-            return (int) invocation.instance.metaClass.getProperty(invocation.instance, 'serverPort')
+            return (int) methodInvocation.instance.metaClass.getProperty(
+                    methodInvocation.instance,
+                    'serverPort'
+            )
         } catch (ignored) {
-            throw new IllegalStateException('Test class must be annotated with @Integration for serverPort to be injected')
+            throw new IllegalStateException(
+                    'Test class must be annotated with @Integration for serverPort to be injected'
+            )
         }
     }
 
     @PackageScope
-    boolean reinitialize(IMethodInvocation invocation) {
-        WebDriverContainerConfiguration specConfiguration = new WebDriverContainerConfiguration(
-                invocation.getSpec()
+    boolean reinitialize(IMethodInvocation methodInvocation) {
+        def specConf = new WebDriverContainerConfiguration(
+                methodInvocation.spec
         )
-        if (matchesCurrentContainerConfiguration(specConfiguration)) {
+        if (matchesCurrentContainerConfiguration(specConf)) {
             return false
         }
 
         if (initialized) {
             stop()
         }
 
-        currentConfiguration = specConfiguration
-        currentContainer = new BrowserWebDriverContainer().withRecordingMode(
-                grailsGebSettings.recordingMode,
-                grailsGebSettings.recordingDirectory,
-                grailsGebSettings.recordingFormat
+        def gebConf = new ConfigurationLoader().conf
+        def dockerImageName = defaultDockerImageName
+
+        def gebConfigFileFound = false
+        if (gebConfigFileExists) {
+            if (!gebConf.driverConf instanceof Closure) {
+                throw new IllegalStateException(
+                        'The "driver" config value must be a Closure that returns an instance of RemoteWebDriver.'
+                )
+            }
+            gebConfigFileFound = true
+
+            // Prepare for creating a suitable container matching the driver
+            // configured in GebConfig.groovy, or more specifically,
+            // specified by the `configuredBrowser` property.
+            dockerImageName = createDockerImageName(gebConf.rawConfig.configuredBrowser)
```

Comment:
regression for GebConfig's that don't have this special var - add a default here?
we already imply Chrome at `new RemoteWebDriver(container.seleniumAddress, new ChromeOptions()` when no GebConfig existed on classpath

Reply:
I didn't mean for us to force an own RemoteWebDriver with ChromeOptions **just** because containerBrowser property is missing, but to (pseudo code - need to find the spot)

```groovy
                } catch (SessionNotCreatedException e){
                    log.error("Failed to create remote browser session: $e.message. Did you set a 'containerBrowser' property corresponding to the driver in GebConfig.groovy?", e)
                }
```

which probably will imply removing the gebConfigFileActive logic